File: frontend/src/modules/categories/services/categoryService.ts
import { categoryApi } from "../api/categoryApi";
import { ICategoryRequest } from "../interfaces/ICategoryRequest";
import { ICategoryService } from "../interfaces/ICategoryService";

export const CategoryService: ICategoryService = {
  CreateCategory: async (category: ICategoryRequest) =>categoryApi.CreateCategory(category),
  GetCategories: async () => categoryApi.GetCategories(),
  GetCategoryById: async (id: number) => categoryApi.GetCategoryById(id),
  UpdateCategory: async (id: number, category: ICategoryRequest) =>categoryApi.UpdateCategory(id, category),
  DeleteCategory: async (id: number) => categoryApi.DeleteCategory(id),
};

File: backend/internal/modules/categories/adapters/services/category_service.go
package services

import (
	"github.com/javierjpv/edenBooks/internal/modules/categories/domain/entities"
	"github.com/javierjpv/edenBooks/internal/modules/categories/domain/ports/repositories"
)

type CategoryService struct {
	repo repositories.CategoryRepository
}

func NewCategoryService(repo repositories.CategoryRepository) *CategoryService {
	return &CategoryService{repo: repo}
}
func (s *CategoryService) CreateCategory(c *entities.Category) error {
	return s.repo.CreateCategory(c)
}

func (s *CategoryService) UpdateCategory(c *entities.Category) error {
	return s.repo.UpdateCategory(c)
}

func (s *CategoryService) DeleteCategory(id uint) error {
	return s.repo.DeleteCategory(id)
}

func (s *CategoryService) GetCategoryByID(id uint) (*entities.Category, error) {
	return s.repo.GetCategoryByID(id)
}

func (s *CategoryService) GetAllCategories() ([]entities.Category, error) {
	return s.repo.GetAllCategories()
}
